packages/vats/src/lib-board.js

```diff
@@ -58,83 +177,161 @@ function makeBoard(
     }
     return Far(`${ifaceInaccessiblePrefix}${iface}`, {});
   };
+  return slotToVal;
+};
 
-  // Create a marshaller that just looks up objects, not publish them.
-  const readonlyMarshaller = Far('board readonly marshaller', {
-    ...makeMarshal(val => {
-      if (!valToId.has(val)) {
-        // Unpublished value.
-        return null;
-      }
-
-      // Published value.
-      return valToId.get(val);
-    }, slotToVal),
-  });
+/**
+ * @typedef {ReturnType<typeof makeMarshal<string>>} StringSlotMarshaller
+ * TODO make Marshall type generic on slot
+ */
 
-  // Create a marshaller useful for publishing all ocaps.
-  const publishingMarshaller = Far('board publishing marshaller', {
-    ...makeMarshal(
-      // Always put the value in the board.
-      // eslint-disable-next-line no-use-before-define
-      val => board.getId(val),
-      slotToVal,
-    ),
-  });
+/**
+ * @param {BoardState} state
+ * @returns {StringSlotMarshaller}
+ */
+const makeReadonlyMarshaller = state => {
+  const { valToId } = state;
+  const slotToVal = makeSlotToVal(state);
+  const valToSlot = val => {
+    if (!valToId.has(val)) {
+      // Unpublished value.
+      return null;
+    }
 
-  const board = Far('Board', {
-    getPublishingMarshaller: () => publishingMarshaller,
-    getReadonlyMarshaller: () => readonlyMarshaller,
-    // Add if not already present
-    getId: value => {
-      if (!valToId.has(value)) {
-        lastSequence += 1n;
-        const seq = lastSequence;
-
-        // Append the CRC, so that the last part of the board ID is
-        // well-distributed.
-        const crcInput = `${prefix}${seq}`;
-        const crc = calcCrc(crcInput, crcDigits);
-
-        const id = `${prefix}${crc}${seq}`;
-
-        valToId.init(value, id);
-        idToVal.init(id, value);
-      }
-      return valToId.get(value);
+    // Published value.
+    return valToId.get(val);
+  };
+  // @ts-expect-error FIXME(Endo) for an unpublished value makeMarshall will push null into slots list
```

Comment:
@dckc the valToSlot here returns null but and makeMarshall treats them as slots anyway:
https://github.com/endojs/endo/blob/686ab61802c0e5206b14f4b91d499e3873555cd4/packages/marshal/src/marshal.js#L93-L96

```js
      index = slots.length;
      const slot = convertValToSlot(passable);
      slots.push(slot);
      slotMap.set(passable, index);
```

Each of the nulls consumes a slot index. I suppose there's some value in indicating that there were unique values, though they've all been omitted. Then this marshaller's slot type isn't really `string` but `string?`. 

WDYT? That would require edits through smart-wallet and CLI but it is more accurate now that I think about it.

Reply:
What do I think of ... which? Changing the type to `string?`? yes, that would make sense.

Reply:
Yes, that. Since it touches so much are you ok with leaving that for another PR?

Reply:
yes.